compiler-core/templates/prelude.js

```diff
@@ -130,6 +130,23 @@ export class BitString {
   get length() {
     return this.buffer.length;
   }
+
+  byteAt(index) {
+    return this.buffer[index];
+  }
+
+  floatAt(index) {
+    return byteArrayToFloat(this.buffer.slice(index, index + 8))
+  }
+
+  intFromSlice(start, end) {
+    return byteArrayToInt(this.buffer.slice(start, end))
+  }
+
+  restFrom(index) {
+    return new BitString(this.buffer.slice(index))
+  }
```

Comment:
Could we get tests for these? https://github.com/gleam-lang/gleam/blob/main/test/javascript_prelude/main.mjs

And type definitions https://github.com/gleam-lang/gleam/blob/main/compiler-core/templates/prelude.d.ts

Can these functions ever raise an exception?